components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FxaClient.kt

```diff
@@ -0,0 +1,423 @@
+     *
+     * @param scopes List of OAuth scopes for which the client wants access
+     * @param entrypoint to be used for metrics
+     * @param metricsParams optional parameters used for metrics
+     * @return String that resolves to the flow URL when complete
+     */
+    @Deprecated("Send beginOAuthFlow to queueAction instead")
+    fun beginOAuthFlow(
+        scopes: Array<String>,
+        entrypoint: String,
+        metricsParams: MetricsParams = MetricsParams(mapOf()),
+    ): String {
+        return this.inner.beginOauthFlow(scopes.toList(), entrypoint, metricsParams)
+    }
+
+    /**
+     * Begins the pairing flow.
+     *
+     * Deprecated: Call `queueAction(FxaAction.BeginPairingFlow(...))` instead.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @param pairingUrl the url to initilaize the paring flow with
+     * @param scopes List of OAuth scopes for which the client wants access
+     * @param entrypoint to be used for metrics
+     * @param metricsParams optional parameters used for metrics
+     * @return String that resoles to the flow URL when complete
+     */
+    @Deprecated("Send beginPairingFlow to queueAction instead")
+    fun beginPairingFlow(
+        pairingUrl: String,
+        scopes: Array<String>,
+        entrypoint: String,
+        metricsParams: MetricsParams = MetricsParams(mapOf()),
+    ): String {
+        return this.inner.beginPairingFlow(pairingUrl, scopes.toList(), entrypoint, metricsParams)
+    }
+
+    /**
+     * Authenticates the current account using the code and state parameters fetched from the
+     * redirect URL reached after completing the sign in flow triggered by [beginOAuthFlow].
+     *
+     * Deprecated: Call `queueAction(FxaAction.CompleteOAuthFlow(...))` instead.
+     *
+     * Modifies the FirefoxAccount state.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     */
+    @Deprecated("Send CompleteOAuthFlow to queueAction instead")
+    fun completeOAuthFlow(code: String, state: String) {
+        this.inner.completeOauthFlow(code, state)
+        this.tryPersistState()
+    }
+
+    // These are used to test error handling in real applications, for example in Firefox Android
+    // with the secret debug menu
+    public fun simulateNetworkError() {
+        Log.w(LOG_TAG, "simulateNetworkError")
+        actionProcessor.simulateNetworkErrorFlag = true
+    }
+
+    public fun simulateTemporaryAuthTokenIssue() {
+        Log.w(LOG_TAG, "simulateTemporaryAuthTokenIssue")
+        inner.simulateTemporaryAuthTokenIssue()
+    }
+
+    public fun simulatePermanentAuthTokenIssue() {
+        Log.w(LOG_TAG, "simulatePermanentAuthTokenIssue")
+        inner.simulatePermanentAuthTokenIssue()
+    }
+}
+
+/**
+ * Implemented by applications to save the Fxa state
+ */
+interface FxaPersistCallback {
+    fun persist(data: String)
+}
```

Comment:
Do we have a naming system for our components?  I tended to prefix everything with `Fxa`, since you don't ever import modules in Kotlin.  I'm open to any suggestions though.